pkg/client/cache/template.go

```diff
@@ -8,7 +8,8 @@ import (
 
 type StoreToTemplateLister interface {
 	List() ([]*templateapi.Template, error)
-	GetTemplateByUID(uid string) (*templateapi.Template, error)
+	ListByNamespace(namespace string) ([]*templateapi.Template, error)
```

Comment:
You should really switch to the generated lister which should already have a per-namespace option.